### PR TITLE
[workspaces] move workspace to other orgs

### PR DESCRIPTION
**Pull request by svenefftinge**
## Description
Allow users to move their workspace to other orgs they are a member of.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - se-move-ws</li>
	<li><b>🔗 URL</b> - <a href="https://se-move-ws.preview.gitpod-dev.com/workspaces" target="_blank">se-move-ws.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Related Issue(s)

Fixes WEB-155

## Release Notes

```release-note
Allow moving workspaces to other organizations
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -1316,6 +1316,36 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         }
     }
 
+    public async moveWorkspace(ctx: TraceContext, workspaceId: string, targetOrganizationId: string): Promise<void> {
```
Comment (easyCZ) [inline components/server/src/workspace/gitpod-server-impl.ts:1319]: This is possibly misleading. It moves the workspace, but it's attribution still remains in the previous org. I wonder if we should even support moving workspaces because it creates this ambiguity about who pays for it after it's been moved

Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: The attribution is on the sessions in the past. Why do you think we should change that after the fact?
Also, why is it ambiguous who pays?

Comment (easyCZ) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: The workspace is attributed the moment it is started, which may belong to org A. I would naturally expect that org A pays for it.

If I now move the workspace to org B, it creates the ambiguity on who actually pays for the workspace (and it's previous instances). Yes, from the code it would always be the org currently selected, but from the user perspective it's not clear and feels like modifying history.

In other words, do we really need to support moving workspaces? Could we instead make it more obvious under which org you're starting a workspace?

Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: I should have mentioned in the description that attribution is the reason we are introducing this feature.
I.e. users have created workspaces on the "wrong" org in the past and are not able to start them now, because that org might not have enough credits. They can move their existing workspaces to the org they wish to get unblocked.

Comment (easyCZ) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: In that case, what if we added a feature which would be "Start this workspace in another org" rather than moving the workspace? This would create a new workspace, but could use the current (must be stopped) one as the basis. We retain immutability this way.

Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: I'm still not sure I understand the issue you see here tbh, let's talk through in our 1on1 today

**Comment by atduarte** [comment]
Should I be seeing something else here? 
image

**Comment by svenefftinge** [comment]
> Should I be seeing something else here?

The action is only visible when you have possible target orgs, the workspace is 'stopped', you are an owner of the containing org.

File: components/server/src/workspace/gitpod-server-impl.ts (hunk 1 above)
Comment (easyCZ) [reply components/server/src/workspace/gitpod-server-impl.ts:1319]: Chatted through it in more detail.

There's an outstanding issue on what happens with workspaces that belong to a project (as a project has project specific env variables, which will also be in snapshots of the workspace).

In general, we agreed that if we can find a way to address the problem without an explicit feature for it (migration), it would be preferable.

**Comment by svenefftinge** [comment]
We decided to not ship this feature as it potentially causes more harm. The situation we wanted to help with is that people would be able to start their workspaces in the org-only mode when the containing org doesn't have any credits. Moving the workspaces to an org that has credits could mess up the state even further as it might not belong their (e.g. the project is configured under the org without credits).
